sympy/holonomic/holonomic.py

```diff
@@ -2185,7 +2183,7 @@ def _find_conditions(simp, x, x0, order, evalf=False):
             else:
                 val = simp.subs(x, x0)
             # return None if it is Infinite or NaN
-            if (val.is_finite is not None and not val.is_finite) or isinstance(val, NaN):
+            if val.is_finite is not None and not val.is_finite or isinstance(val, NaN):
```

Comment:
if val.is_finite is False or isinstance(val, NaN)

Reply:
Makes sense.